### ISSUE TITLE
Remnants of YAML 1.1 Spec in YAML 1.2

### ISSUE THREAD

**Issue by rengzhengcodes** [jbeder/yaml-cpp#1198]
Hello!

I believe there may be an error in how the current codebase handles true/false value resolution for YAML 1.2. The lines in question are linked below:
https://github.com/jbeder/yaml-cpp/blob/0e6e28d1a38224fc8172fae0109ea7f673c096db/src/convert.cpp#L49-L55

According to this section of the YAML 1.2 guide:
https://yaml.org/spec/1.2-old/spec.html#id2805071

Only variants on "true"/"false" in the YAML file should resolve to the booleans. However, the above-linked code also resolves three other ways to represent booleans outside of 1.2 specs but was present in 1.1 specs.

**Comment by jbeder** [comment]
Yep, fair; will accept a PR.

**Comment by stellarpower** [comment]
Is it pretty much as simple as removing those lines? And the question of whether to continue allowing for emission but turn off by default for reading.

I have quickly commented out those lines and added a brief test in my branch [here](https://github.com/stellarpower/yaml-cpp/).

**Comment by perlpunk** [comment]
There are much more differences between YAML 1.1 and 1.2 than just the boolean handling.
See https://perlpunk.github.io/yaml-test-schema/schemas.html where you can find the definitions and test data.
Only changing the boolean handling would implement a YAML version that's neither 1.1 nor 1.2.
And I suppose this should be configurable. Changing it would break things for people wanting to handle 1.1.

**Comment by stellarpower** [comment]
Yes, for sure. I'm consuming yaml-cpp in my project through another package, so don't know much about the present state of play - and ended up here because I have variables in JSON as `"x"` and `"y"` coming out as `"x"` and `true`, so just patched what was blocking me in that branch. Given the readme states the library targets 1.2 I assumed this was a small bug, so was only referencing if any more steps are needed to shave back the tokens interpreted as booleans. It appears to work for now; if reading support for 1.1 is to be dropped or turned off by default, then I guess a larger set of tests for that would be a good place to start.
